lib/skate/detours/trip_modification.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Skate.Detours.TripModification do
+  @moduledoc """
+  A module that produces GTFS-flavored TripModification structs for a detour
+
+  More info: https://gtfs.org/realtime/reference/#message-tripmodifications
+  """
+
+  alias Schedule.Gtfs.RoutePattern
+  alias Schedule.Gtfs.Stop
+
+  defmodule Input do
+    @moduledoc """
+    Input for `Skate.Detours.TripModification.for/1`.
+    """
+
+    @type t :: %__MODULE__{
+            route_pattern: RoutePattern.t(),
+            missed_stops: [Stop.t()]
+          }
+
+    defstruct [:route_pattern, :missed_stops]
+  end
+
+  defmodule SelectedTrip do
+    @moduledoc """
+    Struct representing the `selected_trips` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-selectedtrips
+    """
+
+    @type t :: %__MODULE__{
+            trip_ids: [String.t()]
+          }
+
+    defstruct [:trip_ids]
+  end
+
+  defmodule Modification do
+    @moduledoc """
+    Struct representing the `modifications` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-modification
+    """
+
+    @type t :: %__MODULE__{
+            start_stop_selector: String.t(),
+            end_stop_selector: String.t()
+          }
+
+    defstruct [:start_stop_selector, :end_stop_selector]
+  end
+
+  @type t :: %__MODULE__{
+          selected_trips: [SelectedTrip.t()],
+          modifications: [Modification.t()]
+        }
+  defstruct [:selected_trips, :modifications]
```

Comment:
@mbta/skate-developers this will eventually need to be JSON-encoded, but I'm not familiar enough with `Jason` to know whether that's necessary right now.

...or ever. I found a few other examples of things that get JSON-encoded, where it looks like Jason isn't involved at all, and now I'm wondering if we'll even need to use it here, or if the data structure being right is enough.

Reply:
I _think_ this'll be okay!